app/views/mosaics/show.html.haml

```diff
@@ -11,4 +11,11 @@
               - slice.each_with_index do |color, column|
                 %td{:align => "center"}
                   .block{:id => 8 * row + column, :style => "background-color: #{color};"}
-    %h2 Steps dump: #{@mosaic.steps}
\ No newline at end of file
+    %h2 Steps dump: #{@mosaic.steps}
+    
+    
+    
+    
+-# adding a download button for download.features cucumber tests
+= button_to 'download', mosaic_path(@mosaic), :method => :show
```

Comment:
Does this actually work? You're using the `:show` method, which I don't think is supported by Rails at all. Your path seems to imply that a download route should be:
```ruby
button_to 'download', download_path(@mosaic), :method => :post
```